text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,69 @@
+
+## Explanation
+
+### Maturity levels
+
+Deliverables of a SIG MUST have a declared maturity level, established by SIG maintainers (SIGs), likely with the input of the code owners. While the main deliverable can have a specific maturity level, individual components might have a different one. Examples:
+* the Collector core distribution might declare itself stable and include a receiver that is not stable. In that case, the receiver has to be clearly marked as such
+* the Java Agent might be declared stable, while individual instrumentation packages are not
+
+It is recommended that components should not be marked as stable if their user-visible interfaces are not stable. For instance, if the Collector's component "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" should be at most "beta". Maintainers are free to deviate from this recommendation if they believe users are not going to be affected by future changes.
+
+#### Development
+
+Not all pieces of the component are in place yet, and it might not be available for users yet. Bugs and performance issues are expected to be reported. User feedback is desired, especially regarding the user experience (configuration options, component observability, technical implementation details, ...). Configuration options might break often depending on how things evolve. The component SHOULD NOT be used in production. The component MAY be removed without prior notice.
+
+#### Alpha
+
+This is the default level: any components with no explicit maturity level should be assumed to be "Alpha". The component is ready to be used for limited non-critical production workloads, and the authors of this component welcome user feedback. Bugs and performance problems are encouraged to be reported, but component owners might not work on them immediately. The component's interface and configuration options might often change without backward compatibility guarantees. Components at this stage might be dropped at any time without notice.
+
+#### Beta
+
+Same as Alpha, but the interfaces (API, configuration, generated telemetry) are treated as stable whenever possible. While there might be breaking changes between releases, component owners should try to minimize them. A component at this stage is expected to have had exposure to non-critical production workloads already during its Alpha phase, making it suitable for broader usage.
+
+#### Feature freeze
```

Comment:
I would offer that we name this phase "Stable candidate". "Feature freeze" indicates that we no longer are able to add features, which I don't think is what the text conveys.

Reply:
Would you mind to expand on that? The text does state that this stage means "feature-complete and ready for broader usage", which does imply that no other features will be added until it's declared stable.

Reply:
I will reframe. I don't think having a stage indicating a "feature freeze" is helpful. I am not a partisan of using feature freeze to declare stability.

Reply:
I think what @atoulme is saying is that the important part of this level is the fact that it is "ready to declared stable", not that no new feature will be added to it. I tend to agree with him. I do not see a good reason to prohibit addition of new features to components that are "ready to be declared stable", provided that it does not disrupt their readiness.

So, perhaps we can adopt the name "RC" as it was originally proposed for this level by @reyang [here](https://github.com/open-telemetry/opentelemetry-specification/issues/3459#issuecomment-1530045325)?

Reply:
I'm fine renaming this to Release Candidate, but personally,  RC has the notion of a temporary state that should exist for a few days only, while feature freeze implies a longer period. If that's only my impression, I'll change it to RC, as it's not a strong opinion.

Reply:
IMO, RC vs Feature Freeze don't associate with a particular longer or shorter time period.